File: market/functional-test/src/main/java/ru/yandex/market/FunctionalTest.java
package ru.yandex.market;

import java.io.IOException;
import java.io.InputStream;
import java.nio.charset.StandardCharsets;

import org.apache.commons.io.IOUtils;
import org.apache.commons.lang3.StringUtils;
import org.springframework.core.io.InputStreamResource;
import org.springframework.test.context.ActiveProfiles;

import ru.yandex.market.common.test.db.DbUnitDataSet;
import ru.yandex.market.common.test.junit.JupiterDbUnitTest;

/**
 * Базовый класс для написания функциональных тестов.
 */
@DbUnitDataSet(
        before = "/before.cs_billing.csv",
        dataSource = "csBillingDataSource"
)
@DbUnitDataSet(
        before = "/before.vendors.csv",
        dataSource = "vendorDataSource"
)
@ActiveProfiles({"functionalTest", "development"})
public abstract class FunctionalTest extends JupiterDbUnitTest {
    protected InputStreamResource getInputStreamResource(String filename) {
        return new InputStreamResource(getClass().getResourceAsStream(
                getClass().getSimpleName() + filename));
    }

    protected String getStringResource(String name) {
        Class<? extends FunctionalTest> cls = getClass();
        return readFromFile(cls, cls.getSimpleName() + name);
    }

    private static String readFromFile(Class<?> contextClass, String jsonFileName) {
        try (InputStream in = contextClass.getResourceAsStream(jsonFileName)) {
            final String resource = IOUtils.toString(in, StandardCharsets.UTF_8);
            return StringUtils.trimToNull(resource);
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}

File: market/test/kotlin/ru/yandex/market/mbi/marketing/FunctionalTest.kt
package ru.yandex.market.mbi.marketing

import org.apache.commons.io.IOUtils
import org.apache.commons.lang3.StringUtils
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.web.server.LocalServerPort
import org.springframework.core.io.InputStreamResource
import org.springframework.test.context.ActiveProfiles
import ru.yandex.market.common.test.junit.JupiterDbUnitTest
import ru.yandex.market.mbi.marketing.config.MbiMarketingFunctionalTestConfig
import java.io.IOException
import java.nio.charset.StandardCharsets

@SpringBootTest(
    properties = ["spring.main.allow-bean-definition-overriding=true"],
    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
    classes = [MbiMarketingFunctionalTestConfig::class]
)
@ActiveProfiles("functionalTest")
open class FunctionalTest : JupiterDbUnitTest() {

    @LocalServerPort
    protected var serverPort = 0

    protected open fun baseUrl(): String {
        return "http://localhost:$serverPort"
    }

    protected fun getInputStreamResource(filename: String): InputStreamResource {
        return InputStreamResource(
            javaClass.getResourceAsStream(
                javaClass.simpleName + filename
            )
        )
    }

    protected fun getStringResource(name: String): String? {
        val cls: Class<out FunctionalTest?> = javaClass
        return readFromFile(cls, cls.simpleName + name)
    }

    private fun readFromFile(contextClass: Class<*>, jsonFileName: String): String? {
        try {
            contextClass.getResourceAsStream(jsonFileName).use { `in` ->
                val resource = IOUtils.toString(`in`, StandardCharsets.UTF_8.name())
                return StringUtils.trimToNull(resource)
            }
        } catch (e: IOException) {
            throw RuntimeException(e)
        }
    }
}
